src/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/pipeline/NoopListener.java

```diff
@@ -9,13 +9,13 @@
  */
 public class NoopListener implements Listener{
     private static final Logger LOGGER = LoggerFactory.getLogger(NoopListener.class);
-    private Launcher launcher;
+    private LauncherAPI launcher;
```

Comment:
Concrete-to-interface, that's one of the best refactoring ever!

Reply:
Dependency inversion principle :)